### PR TITLE
Support latest babel 7 only

### PR DESCRIPTION
**Pull request by danez**
This drops babel 6 as babel 7 now has also different package names which makes it rather complicated to support both.

This fixes several issues:
 - The plugins received from OptionManager are now classes and not arrays and have a prop `options` 
- The plugin key that the resolver was checking in the babel 7 case was wrong it is now `module-resolver` see here https://github.com/tleunen/babel-plugin-module-resolver/blob/master/src/index.js#L25
- The tests are broken, because jest by default sets the env to "test" which was interfering with the test that tested different environments.
- The babel internal test does not work as internals are not exposed this way anymore

This is basically the same as #76 with fixed tests.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -36,12 +36,13 @@
     "resolve": "^1.4.0"
   },
   "devDependencies": {
-    "babel-cli": "^7.0.0-beta.2",
-    "babel-core": "^7.0.0-beta.2",
+    "@babel/cli": "^7.0.0-beta.34",
+    "@babel/core": "^7.0.0-beta.34",
+    "@babel/plugin-proposal-object-rest-spread": "^7.0.0-beta.34",
+    "@babel/preset-env": "^7.0.0-beta.34",
+    "babel-core": "^7.0.0-0",
```
Comment (danez) [inline package.json]: This is needed for jest and installs the babel-core bridge. https://github.com/babel/babel-bridge
We could have used the bridge here, but I think creating a new major version is not that big of a problem here, than it is in jest.

Comment (tleunen) [reply package.json]: Oh nice! I noticed that issue with jest when I worked on that in #76 

Thank you so much for fixing it.

**Comment by tleunen** [comment]
What do you suggest here? Moving `@babel/core` from peerDep to dep? Or releasing a new major version?

Since it's still a beta, most users are probably still using babel 6 and I'm guessing this might create more issues for them?

**Comment by danez** [comment]
I'm not sure. I think I would release 4.0 soonish, as it is needed for the babel plugin 3.x?

To not break more users i think 4.0 should be released as is. Or with babel 7 removed as it is only working with some older babel 7 version anyway. If not removed I would only fix the [key check](https://github.com/tleunen/eslint-import-resolver-babel-module/blob/master/src/index.js#L19) for babel 7, as it is obviously wrong now that the babel plugin sets its name [here](https://github.com/tleunen/babel-plugin-module-resolver/commit/ba978d8f18b8605b83547de2009d9ee836c92796).

But I guess I would prefer releasing 4.0 without babel 7 and then do 5.x-beta for babel 7.

Complete other solution would be to support both, same as jest. But then users of babel 7 have to install @babel/core@7 and babel-core@7.0.0-bridge.0.  see https://facebook.github.io/jest/docs/en/getting-started.html#using-babel

**Comment by tleunen** [comment]
Yes good idea. I'll release v4 with Babel 6, then v5 with Babel 7.

Forcing the users to install 7 and the babel-core bridge is not a viable solution imo. It's a bit an ugly hack :/

(I'll actually finally try to setup these 2 packages with lerna and keep the versions synchronized to make it easier for all users)

**Pull request merged**